receiver/prometheusreceiver/DESIGN.md

```diff
@@ -126,8 +126,8 @@ shown in the flowchart below:
 
 It basically does the following things in turn:
 
-  1. make a http call to fetch data from the binding [target](#target)'s metrics endpoint with [scraper](#scraper)
-  2. acquired a [storage appender](#storage-appender) instance with the [Appendable](#appendable) interface 
+  1. make a http call to fetch data from the binding [target](https://github.com/prometheus/prometheus/blob/v2.9.2/scrape/target.go)'s metrics endpoint with [scraper](https://github.com/prometheus/prometheus/blob/d3245f15022551c6fc8281766ea62db4d71e2747/scrape/scrape.go#L506-L511)
+  2. acquired a [storage appender](https://github.com/prometheus/prometheus/blob/d3245f15022551c6fc8281766ea62db4d71e2747/storage/interface.go#L86-L95) instance with the [Appendable](https://github.com/prometheus/prometheus/blob/d3245f15022551c6fc8281766ea62db4d71e2747/scrape/manager.go#L37-L39) interface 
```

Comment:
As there any documentation sources we can use instead of the go code?

Reply:
Probably don't have. the `[target]`  or `[storage appender]` are most relate to this section:
```
### Major components of Prometheus Scrape package

- **[ScrapeManager](https://github.com/prometheus/prometheus/blob/v2.9.2/scrape/manager.go):**
the component which loads the scrape_config, and manages the scraping tasks

- **[ScrapePool](https://github.com/prometheus/prometheus/blob/d3245f15022551c6fc8281766ea62db4d71e2747/scrape/scrape.go#L154-L439):**
an object which manages scrapes for a sets of targets

- **[Scraper](https://github.com/prometheus/prometheus/blob/d3245f15022551c6fc8281766ea62db4d71e2747/scrape/scrape.go#L506-L511):**
a http client to fetch data from remote metrics endpoints

- **[Target](https://github.com/prometheus/prometheus/blob/v2.9.2/scrape/target.go):**
the remote metric endpoint, as well as related relabeling settings and other metadata
```

maybe we can add a subtitle for all of components in the section, then we can use documentation sources

Reply:
I like this idea.